python/cudf/cudf/tests/test_binops.py

```diff
@@ -772,3 +871,584 @@ def test_ufunc_ops(lhs, rhs, ops):
         utils.assert_eq(
             expect, got,
         )
+
+
+
+
+def dtype_scalar(val, dtype):
+    dtype = np.dtype(dtype)
+    if dtype.type in {np.datetime64, np.timedelta64}:
+        res, _ = np.datetime_data(dtype)
+        return dtype.type(val, res)
+    else:
+        return dtype.type(val)
+
+
+def make_valid_scalar_add_data():
+    valid = set()
+
+    # to any int, we may add any kind of
+    # other int, float, datetime timedelta, or bool
+    valid |= set(
+        product(
+            INTEGER_TYPES,
+            FLOAT_TYPES | DATETIME_TYPES | TIMEDELTA_TYPES | BOOL_TYPES,
+        )
+    )
+
+    # to any float, we may add any int, float, or bool
+    valid |= set(
+        product(FLOAT_TYPES, INTEGER_TYPES | FLOAT_TYPES | BOOL_TYPES)
+    )
+
+    # to any datetime, we may add any int, timedelta, or bool
+    valid |= set(
+        product(DATETIME_TYPES, INTEGER_TYPES | TIMEDELTA_TYPES | BOOL_TYPES)
+    )
+
+    # to any timedelta, we may add any int, datetime, other timedelta, or bool
+    valid |= set(
+        product(TIMEDELTA_TYPES, INTEGER_TYPES | DATETIME_TYPES | BOOL_TYPES)
+    )
+
+    # to any bool, we may add any int, float, datetime, timedelta, or bool
+    valid |= set(
+        product(
+            BOOL_TYPES,
+            INTEGER_TYPES
+            | FLOAT_TYPES
+            | DATETIME_TYPES
+            | TIMEDELTA_TYPES
+            | BOOL_TYPES,
+        )
+    )
+
+    # to any string, we may add any other string
+    valid |= {("str", "str")}
+
+    return valid
+
+
+def make_invalid_scalar_add_data():
+    invalid = set()
+
+    # we can not add a datetime to a float
+    invalid |= set(product(FLOAT_TYPES, DATETIME_TYPES))
+
+    # We can not add a timedelta to a float
+    invalid |= set(product(FLOAT_TYPES, TIMEDELTA_TYPES))
+
+    # we can not add a float to any datetime
+    invalid |= set(product(DATETIME_TYPES, FLOAT_TYPES))
+
+    # can can not add a datetime to a datetime
+    invalid |= set(product(DATETIME_TYPES, DATETIME_TYPES))
+
+    # can not add a timedelta to a float
+    invalid |= set(product(FLOAT_TYPES, TIMEDELTA_TYPES))
+
+    return invalid
+
+
+@pytest.mark.parametrize("dtype_l,dtype_r", make_valid_scalar_add_data())
```

Comment:
I think we need to convert sets into a stable-ordered list to make it safe for `pytest-xdist`